lib/seahorn/BvOpSem2MemRepr.cc

```diff
@@ -14,11 +15,93 @@ auto as_std_array(const T &t, const Rest &... rest) ->
 namespace seahorn {
 namespace details {
 
-OpSemMemRepr::MemValTy OpSemMemArrayRepr::MemSet(PtrTy ptr, Expr _val,
-                                                 unsigned len, MemValTy mem,
-                                                 unsigned wordSzInBytes,
-                                                 PtrSortTy ptrSort,
-                                                 uint32_t align) {
+/**
+ * rewriter for store expr:
+ * e = store A idx val -> { return ite( (idx == ptr), val, rewrite(A) )}
+ * e is not store (const array or terminal register) ->
+ * { return select(e, m_ptr) }
+ **/
+struct ArrayStoreRewriter : public std::unary_function<Expr, Expr> {
+  Expr m_ptr;
+  OpSemAlu &m_alu;
+  unsigned m_ptrSz;
+  ArrayStoreRewriter(Expr ptr, OpSemAlu &alu, unsigned ptrSz)
+      : m_ptr(ptr), m_alu(alu), m_ptrSz(ptrSz) {}
+
+  Expr doPtrEq(Expr p1, Expr p2) { return m_alu.doEq(p1, p2, m_ptrSz); }
+
+  Expr operator()(Expr e) {
+    if (isOp<STORE>(e)) {
```

Comment:
will adding an `assert` that we expect a struct with three fields help with debugging?